docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,488 @@
+  there is an output parameter to indicate the specific error code. In this case,
+  the verifier can decide whehter the SGX quote is valid based on their own policy.
+
+We propose to introduce a new SGX-specific claim `OE_CLAIM_SGX_QUOTE_VERIFICATION_STATUS` to
+indicate QVL/QvE return status. The verifier can check the status to see if there
+is a non-fatal error during quote verification. This claim can have one of three values:
+
+  - OE_OK - Quote verification passed, and TCB level is up to date
+  - Non-Fatal error code - API returns OE_OK,
+    but claim `OE_CLAIM_SGX_QUOTE_VERIFICATION_STATUS` in claims
+    returns an error code. The verifier can refer to this status in his/her
+    verification policy. Non-Fatal error can be one of below:
+
+```C
+   OE_SGX_CONFIG_NEEDED // Quote verification passed and the platform is patched
+                        // to latest TCB level, but additional configuration of
+                        // the SGX platform may be neede. For example, attester
+                        // may need to enable/disable SMT in BIOS
+   OE_SGX_OUT_OF_DATE  // The Quote is good but TCB level of the platform is
+                       // out of date, additional configuration of the SGX
+                       // Platform at its current patching level may be needed.
+                       // The platform needs patching to be at the latest TCB level
+   OE_SGX_SW_HARDENING_NEEDED  // The TCB level of the platform is up to date,
+                               // but SGX SW Hardening is needed. For example,
+                               // attester may need to apply LVI mitgation in
+                               // his/her enclave by linking LIV version SDK
+   OE_SGX_CONFIG_AND_SW_HARDENING_NEEDED  // The TCB level of the platform is up
+                                          // to date, but additional
+                                          // configuration of the platform at
+                                          // its current patching level may be
+                                          // needed. Moreove, SGX SW Hardening
+                                          // is also needed. For example,
+                                          // attester may need enable/disable
+                                          // SMT in BIOS, also he/she needs to
+                                          // apply LVI mitgation in enclave
+
+Note: SGX TCB = Platform CPU HW/FW TCB + SGX PCE TCB.
+```
+If the verifier needs to provide a different quote verification policy
+beyond the policy enforced by the sgx_qv_verify_quote() API, the verifier can
+request sgx_qv_verify_quote() API to return supplemental data. The detailed
+supplemental data definition are described in above `Background` section.
+
+So we also propose to add SGX supplemental data fields the the list of claims.
+
+[PR 3353](https://github.com/openenclave/openenclave/pull/3353) added SGX
```

Comment:
There is at least one customer who is interested in raw TCB Info in JSON format. It may not be a hard requirement.

Reply:
May or May not?
If it's a hard requirement to return raw collateral data, then I need to consider the alternatives.

Reply:
It may not. Let me get back to you on this.

Reply:
One question would be what this customer wants the raw TCB info blob for? If their real requirements are the individual fields in this blob that we will provide as a set of claims, then this proposal would meet their requirements better.

Reply:
It's a hard requirement but I understand in step 1 the raw TCB Info will be kept.